app/models/html_file.rb

```diff
@@ -632,43 +631,85 @@ def create_WEM_new(person_id, the_title, the_markdown, multiple, pub_status = ni
     if self.status == 'Accepted'
       begin
         tt = the_title.strip
-        p = Person.find(person_id)
-        Chewy.strategy(:atomic) {
+        p = Person.find(person_id) # TODO: use self.person_id instead of argument? (Damir)
+        Chewy.strategy(:atomic) do
           ActiveRecord::Base.transaction do
-            w = Work.new(title: tt, orig_lang: orig_lang, genre: genre, comment: comments, primary: true) # TODO: un-hardcode primariness in new upload flow
+            w = Work.new(
+              title: tt,
+              orig_lang: orig_lang,
+              genre: genre,
+              comment: comments,
+              primary: true # TODO: un-hardcode primariness in new upload flow
+            )
+
             q = (translator_id.nil? ? p : translator)
-            copyrighted = ((p.public_domain && q.public_domain) ? false : ((p.public_domain.nil? || q.public_domain.nil?) ? nil : true)) # if author is PD, expression is PD # TODO: make this depend on both work and expression author, for translations
-            e = Expression.new(title: tt, language: 'he', period: q.period, copyrighted: copyrighted, source_edition: publisher, date: year_published, comment: comments) # ISO codes
-            w.expressions << e
-            w.save!
-            c = Creation.new(work_id: w.id, person_id: p.id, role: :author)
-            c.save!
+            intellectual_property =
+              if p.intellectual_property_copyrighted? || q.intellectual_property_copyrighted?
```

Comment:
I believe this is a correct way to determine intellectual_property type based on values from author and translator.